diff --git a/src/pc/djui/djui_panel_pause.c b/src/pc/djui/djui_panel_pause.c
index 6a4c2ef..1b9f453 100644
--- a/src/pc/djui/djui_panel_pause.c
+++ b/src/pc/djui/djui_panel_pause.c
@@ -52,6 +52,17 @@ static void djui_panel_pause_quit(struct DjuiBase* caller) {
     }
 }
 
+static void djui_panel_pause_exit_tater_tube_yes(UNUSED struct DjuiBase* caller) {
+    game_exit();
+}
+
+static void djui_panel_pause_exit_tater_tube(struct DjuiBase* caller) {
+    djui_panel_confirm_create(caller,
+                              "EXIT TO TATER TUBE",
+                              "RETURN TO TATER TUBE?",
+                              djui_panel_pause_exit_tater_tube_yes);
+}
+
 void djui_panel_pause_create(struct DjuiBase* caller) {
     if (gDjuiPanelPauseCreated) { return; }
     if (gDjuiChatBoxFocus) { djui_chat_box_toggle(); }
@@ -114,6 +125,8 @@ void djui_panel_pause_create(struct DjuiBase* caller) {
         } else {
             djui_button_create(body, DLANG(PAUSE, DISCONNECT), DJUI_BUTTON_STYLE_BACK, djui_panel_pause_quit);
         }
+
+        djui_button_create(body, "EXIT TO TATER TUBE", DJUI_BUTTON_STYLE_BACK, djui_panel_pause_exit_tater_tube);
     }
 
     djui_panel_add(caller, panel, defaultBase);
diff --git a/src/pc/gfx/gfx_opengl.c b/src/pc/gfx/gfx_opengl.c
--- a/src/pc/gfx/gfx_opengl.c
+++ b/src/pc/gfx/gfx_opengl.c
@@ -834,11 +834,13 @@ static void gfx_opengl_init(void) {
     glGenBuffers(1, &opengl_vbo);
 
     glBindBuffer(GL_ARRAY_BUFFER, opengl_vbo);
 
+#ifndef USE_GLES
     if (vmajor >= 3 && !is_es) {
         glGenVertexArrays(1, &opengl_vao);
         glBindVertexArray(opengl_vao);
     }
+#endif
 
     glDepthFunc(GL_LEQUAL);
     glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
